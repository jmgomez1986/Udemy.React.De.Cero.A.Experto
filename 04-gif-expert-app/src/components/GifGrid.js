import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

	const [images, setImages] = useState([]);

	useEffect(() => {
		getGifs();
	}, []); // Al mandar la lista  de dependencias vacia, se ejecta solo en la primera renderizacion

	const getGifs = async () => {
		const url = `https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=5Aur2asApFfGBynHWjHEVbAzIUjLAKK5`
		const response = await fetch(url);
		const {data} = await response.json();

		const gifs = data.map( (img) => {
			return {
				id: img.id,
				title: img.title,
				url: img.images?.downsized_medium.url
			}
		});

		setImages(gifs);
	}

	return (
		<>

			<h3>{category}</h3>

			<div className="card-grid">
				
				{
					images.map( (img) => 
						<GifGridItem
							key={img.id}
							// img={img}
							{...img}
						/>
					)
				}

			</div>
			
		</>
	)
}
