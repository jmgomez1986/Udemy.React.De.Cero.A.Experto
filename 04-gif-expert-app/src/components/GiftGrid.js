import React from 'react'

export const GiftGrid = ({category}) => {

	const getGifs = async () => {
		const url = `https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=5Aur2asApFfGBynHWjHEVbAzIUjLAKK5`
		const response = await fetch(url);
		const {data} = await response.json();

		const gifs = data.map( (img) => {
			return {
				id: img.id,
				title: img.title,
				url: img.images?.downsized_medium.url
			}
		})

		console.log(gifs);
	}

	getGifs();

	return (
		<div>
			<h3>{category}</h3>
		</div>
	)
}
