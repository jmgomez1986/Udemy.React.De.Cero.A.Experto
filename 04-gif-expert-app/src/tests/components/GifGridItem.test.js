import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Prueba en componente <GifGridItem />', () => {	
	const title = 'Un titulo';
	const url = 'https://localhost/algo.jpg';
	const wrapper = shallow(
		<GifGridItem
			title={title}
			url={url}
		/>);

	test('debe de mostrar el componente correctamente ', () => {
			expect(wrapper).toMatchSnapshot();
	});

	test('debe de tener un parrafo con el titulo ', () => {
		const gifGridItemTitle = wrapper.find('p').text().trim();

		expect(gifGridItemTitle).toBe(title);
	});
	
	test('debe de tener una imagen con la url y alt igual a los props ', () => {
		const gifGridItemImg = wrapper.find('img');

		expect(gifGridItemImg.prop('src')).toBe(url);
		expect(gifGridItemImg.prop('alt')).toBe(title);

		// const gifGridItemImg = wrapper.find('img').props();

		// expect(gifGridItemImg.src).toBe(url);
		// expect(gifGridItemImg.alt).toBe(title);
	});

	test('debe tener la clase animate__fadeIn ', () => {
		const div = wrapper.find('div');
		const className = div.props().className;

		expect(className.includes('animate__fadeIn')).toBe(true);
	});	
	
});
