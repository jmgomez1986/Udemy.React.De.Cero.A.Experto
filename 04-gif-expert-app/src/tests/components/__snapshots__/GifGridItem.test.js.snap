// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente <GifGridItem /> debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;

exports[`Prueba en componente GifGridItem debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
